Add What-We-Use file in tooling [SATURN-1733]

Shakespeared opened the pull request:
Add master list of 'what we use' as a directory of docs and links in the tooling section. Definitely open to comments/suggestions about if this is the right format or organization, just wanted to get something in there.

Shakespeared committed:
Add What-We-Use file in tooling

Add master list of 'what we use' as a directory of docs and links in the tooling section. Definitely open to comments/suggestions about if this is the right format or oranization, just wanted to get something in there.

ehanna4 commented on tooling/What-We-Use.md:
```diff
@@ -0,0 +1,20 @@
+|Github/Github desktop|Version control|Code|
+|Google calendar|Scheduling/Shared calendars|Team|
+|IntelliJ|Code editor|Code|
+|Jira|Sprint kanban board|Scrum|
```
The Kanban board is not currently in use so I would recommend removing 'kanban' from the description here, and maybe say like 'Sprint board and backlog' instead?

Shakespeared replied:
Is Jira itself not a Kanban board? I'd always heard "Kanban" as a general term for that kind of ticketing system/display board, but maybe that's not right. I agree that having slightly more detail in the description would be good.

ehanna4 replied:
Kanban and Scum are two styles of Agile, and Jira supports boards for both of these. We had a Kanban board for awhile and I think it's still linked in our Jira Project which is why it might be confusing to have Kanban in the description for Jira, as we should only be using the Sprint/Scrum board at this point.

Shakespeared replied:
Gotcha, I didn't realize it had a more specific usage! I'll update that.

s-rubenstein reviewed (commented):
I like this idea, but it kinda feels like it overlaps with our Table of Contents. Do you think this provides new information on top of that?

ehanna4 commented on tooling/What-We-Use.md:
```diff
@@ -0,0 +1,20 @@
+|Vault|Key/Secret storage|Code|
+|Lodash|Js library|Code|
+|Hyperscript Helper|Js extension|Code|
+
```
Two more additions which I'm not sure are fully team sanctioned but I have seen at least a few members use are: Insomnia and ITerm2

Shakespeared replied:
@blrubenstein Yeah there is definitely an argument for merging this with the TOC. My thought process was that it might be nice to have it in the tooling directory for clarity, but we could also just make a 'tooling' section of the TOC, which now that I'm thinking of it, might be neater overall. 

We could also have different sections for 'team tools' and 'optional tools,' thoughts about that?

Shakespeared replied:
Made a new commit with some updates, I moved the 'what we use' list to the readme/TOC, which I think is better, and a few other small changes. Thanks for the suggestion on that!

Shakespeared committed:
move tool list to readme, formatting changes

move the tool list to the readme/TOC doc instead of its own file. Also change the list formatting on the whos-who doc to pure markdown, and added a file for slack channels for new team members to join.

Let me know if this organization seems better.

ehanna4 commented on tooling/Slack-channels.md:
```diff
@@ -0,0 +1,17 @@
+
+##Join slack: broadinstitute.slack.com
+
+Slack is a major communication method for the team for questions and discussion. Saturn-team is the primary team channel.
```
I think some sort of clarifiction about #saturn vs #saturn-team would be good here. Something like:

"#saturn-team is the private team channel for team discussion and alerting other members about absences etc. while #Saturn is a public channel where other teams can bring questions to the Saturn team."

Shakespeared replied:
I added descriptions to the #saturn and #saturn-team channels, as well as a few others. Some of them I wasn't quite clear enough on the details to annotate (e.g. the nuances of #dsp-engineering, and #dsp-workbench, and the channel descriptions aren't very helpful). I put a TODO on it, but feel free to put suggestions here or in slack and I can drop them in if you want.

ehanna4 commented on tooling/Slack-channels.md:
```diff
@@ -0,0 +1,17 @@
+
+Recommended channels: 
+
+* Saturn-team
```
also just to be nit-picky I would recommend making 'Saturn-team' and 'Saturn' lowercase as that's slack's convention and the other channels in this list are lowercased

Shakespeared replied:
Thanks for this, that's a good call! I'll change the formatting and add some descriptions for the channels.

s-rubenstein commented on tooling/Slack-channels.md:
```diff
@@ -0,0 +1,17 @@
+* prometheus
+* softeng
+* 105b
+* social
```
Optional: It may be worth adding `workbench-userliaison` to this list? (I mention it in the User Liaison doc, so, probably not necessary since you should only need to watch the channel if you are in that role.)

Shakespeared replied:
That makes sense, I'll add it! I put in a quick description as well, but you should take a look to make sure it's accurate/complete since I'm not too familiar with the user liaison procedure.

ehanna4 replied:
If you're adding `workbench-userliaison` I would recommend maybe adding like a "priveleged" or "extra" section and include that and also the on-call channels ( I will get those for you in a sec)

Shakespeared replied:
Maybe I should break it out into sections, like 'team/work channels' 'social channels' 'extra channels' (I'm not sure 'extra' is the best descriptor but I can't think of what else to call it).

ehanna4 replied:
yes I like that idea !

Shakespeared committed:
update slack-channels.md

Update formatting, add #workbench-userliaison channel, add descriptions to some channels.

ehanna4 commented on tooling/Slack-channels.md:
```diff
@@ -0,0 +1,23 @@
+|\#dsp-engineering|
+|\#dsp-workbench|
+|\#workbench-userliaison|channel for the current user liason to watch for questions and issues|
+|\#prometheus|
```
```suggestion
|\#prometheus|channel for DSP coffee talk|
```

ehanna4 commented on tooling/Slack-channels.md:
```diff
@@ -0,0 +1,23 @@
+|\#workbench-userliaison|channel for the current user liason to watch for questions and issues|
+|\#prometheus|
+|\#softeng|discussions on software/engineering
+|\#105b|
```
```suggestion
|\#105b| channel for building related news and updates |
```

Shakespeared committed:
Merge branch 'dev' into kf-add-new-files

Shakespeared committed:
update slack-channels.md

update the list of slack channels so it can be integrated into the other communication/social docs

ehanna4 committed:
Update tooling/Slack-channels.md

Shakespeared committed:
update slack-channels.md descriptions

update a few channel descriptions I missed before

Shakespeared committed:
Merge branch 'kf-add-new-files' of https://github.com/DataBiosphere/saturn-documentation into kf-add-new-files

Shakespeared committed:
Merge branch 'dev' into kf-add-new-files

s-rubenstein commented on tooling/Slack-channels.md:
```diff
@@ -0,0 +1,38 @@
+# Slack
```
Nit: I'd rename the file `Slack.md` Just given that it covers more than just the channels.

s-rubenstein reviewed (approved):
LGTM Pending one nit.

Shakespeared committed:
update slack file name

update file name from slack-channels.md to slack.md
